docs/api/caching.md

```diff
@@ -35,3 +36,252 @@ memory, letting you pick up where you left off.
     `**/__marimo__/cache/` to your `.gitignore`.
 
 ::: marimo.persistent_cache
+
+## Caching expectations
+
+The performance and reliability of your notebook may depend on the caching functionality—as such, it is important to understand the expectations of marimo's caching utilities.
+
+`mo.cache` context blocks are designed to provide deterministic cache restoration when using marimo's APIs for state management.
+
+- On cache hit, the cached values are restored and the code block is not executed, meaning no side effects like printing, file I/O, or network requests will occur.
+- Rerunning a cell will not remove your cache, enabling cache hits on subsequent runs.
+- Dynamic changes to the notebook will appropriately invalidate the cache, if either:
+    1. the code in the defined block changes,
+    2. the variables with basic data types used by your code block change,
```

Comment:
i think it would be good to provide examples for 2 and 3.

e.g. `x = 3`
and 
e.g. `pd.read_csv(https://some-external-url)

Reply:
In coming steps I have an idea for how to make this more explicit, even in the example you provided

```pythonl
data = pd.read_csv(https://some-external-url/)
```

Should not rely on code because pandas is "hashable". But we'll add this in once we provide an API to check